storage/innobase/fil/fil0fil.cc

```diff
@@ -11344,15 +11344,14 @@ byte *fil_tablespace_redo_encryption(byte *ptr, const byte *end,
         exit(EXIT_FAILURE);
       }
       return (ptr + len);
-
-    } else {
-      ulint master_key_id = mach_read_from_4(ptr + Encryption::MAGIC_SIZE);
-      if (Encryption::get_master_key_id() < master_key_id) {
-        Encryption::set_master_key(master_key_id);
-      }
-      bool found = xb_fetch_tablespace_key(space_id, key, iv);
-      ut_a(found);
     }
+  } else {
+    ulint master_key_id = mach_read_from_4(ptr + Encryption::MAGIC_SIZE);
+    if (Encryption::get_master_key_id() < master_key_id) {
+      Encryption::set_master_key(master_key_id);
+    }
+    bool found = xb_fetch_tablespace_key(space_id, key, iv);
+    ut_a(found);
   }
```

Comment:
Previously, the code says, after decode encryption is successful, we will use xb_fetch_tablespace_key(). Which in turn looks for prefedeined xtrabackup_keys. heh? we do why do this? :thinking: 

I don't understand this change as well. Looks more dangerous to use this without the decoding.

Reply:
this is previous [code](https://github.com/percona/percona-xtrabackup/blob/3144bd4c22cced64cc5444b81f7f82187c938a13/storage/innobase/fil/fil0fil.cc#L11222)

I split change in two commits. that's why it look confusing.  xb_fetch_tablespace_key is done with transition-key in that case we don't do decode. we rely on xb_fetch_tablespace_key.

Reply:
Ok, so the conflict resolution moved it to wrong place and this commit moves it to the original place again? (Essentially we brought it to pre-merge state?)?

Reply:
yes, we brought the logic to pre conflict state.

Reply:
ok, got it